src/coreclr/System.Private.CoreLib/src/System/Runtime/InteropServices/Marshal.CoreCLR.cs

```diff
@@ -445,6 +460,11 @@ public static object GetUniqueObjectForIUnknown(IntPtr unknown)
         [SupportedOSPlatform("windows")]
         public static int ReleaseComObject(object o)
         {
+            if (!IsComSupported)
+            {
+                throw new NotSupportedException(SR.NotSupported_COM);
+            }
+
```

Comment:
Wouldn't this work with ComWrappers - I thought that AddRef/Release is handled by the runtime in both cases...

Reply:
Nope, the __ComObject check below apparently is only for built-in COM objects